src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +293,57 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            int extCount = 0;
+            try
+            {
+                //register the new assembly
+                var serviceProvider = (ExtensionServiceProvider)ServiceHostInstance.ServiceProvider;
+                var assemblies = new Assembly[] { AssemblyLoadContext.Default.LoadFromAssemblyPath(param.Assembly) };
+                serviceProvider.AddAssembliesToConfiguration(assemblies);
+
+                //only load the completion providers from the new assembly
+                serviceProvider = ExtensionServiceProvider.Create(assemblies);
+                foreach (var provider in serviceProvider.GetServices<ICompletionExtensionProvider>())
+                {
+                    var cancellationTokenSource = new CancellationTokenSource();
+                    cancellationTokenSource.CancelAfter(ExtensionLoadingTimeout);
+                    var cancellationToken = cancellationTokenSource.Token;
+                    var ext = await provider.CreateAsync(param.Properties ?? new Dictionary<string, object>(), cancellationToken).WithTimeout(ExtensionLoadingTimeout);
+                    if (ext == null)
+                    {
+                        await requestContext.SendError("Failed to create ICompletionExtension");
+                        continue;
+                    }
+                    string extName = ext.Name;
+                    await ext.Initialize(cancellationToken).WithTimeout(ExtensionLoadingTimeout);
+                    cancellationTokenSource.Dispose();
+                    if (!string.IsNullOrEmpty(extName))
+                    {
+                        _completionExtensions.AddOrUpdate(extName, ext, (_, previous) =>
+                        {
+                            previous?.Dispose();
+                            return ext;
+                        });
+                        extCount++;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                await requestContext.SendError(ex.Message);
+            }
```

Comment:
Should return here otherwise we could end up sending both an error and a result - but only one of those should be expected

Reply:
@kevcunnane To verify - but my understanding is that an error is a response that has additional error metadata (which IIRC is actually turned into an exception on the ADS side). We should only ever send one response for a single request so sending both an error response and a normal response isn't going to work correctly and will likely cause another error.

Reply:
Yes, we should return here and only send result on success. Also re-reviewing, there's a line that sends error if 1 provider is null and then continues, that's a bug too. 

@shengyfu - please verify that max 1 `SendError` or `SendResult` is ever sent. Otherwise we'll get unexpected behavior.

Reply:
Yes, fixed now.